dag/dag.go

```diff
@@ -346,6 +341,11 @@ func (g *AcyclicGraph) ReverseDepthFirstWalk(start []Vertex, f DepthWalkFunc) er
 				Depth:  current.Depth + 1,
 			})
 		}
+
+		// Visit the current node
+		if err := f(current.Vertex, current.Depth); err != nil {
+			return err
+		}
```

Comment:
The other graph walk has this same problem, but I wanted to address it in a separate PR w/ its own tests.